proto/internal/temporal/server/api/matchingservice/v1/request_response.proto

```diff
@@ -190,8 +190,31 @@ message DescribeTaskQueueRequest {
 }
 
 message DescribeTaskQueueResponse {
-    repeated temporal.api.taskqueue.v1.PollerInfo pollers = 1;
-    temporal.api.taskqueue.v1.TaskQueueStatus task_queue_status = 2;
+    temporal.api.workflowservice.v1.DescribeTaskQueueResponse desc_response = 1;
+}
+
+message DescribeTaskQueuePartitionRequest {
+    string namespace_id = 1;
+    temporal.server.api.taskqueue.v1.TaskQueuePartition task_queue_partition = 2;
+
+    // If not provided, the result for the default Build ID will be returned. The default Build ID is the one
+    // mentioned in the first unconditional Assignment Rule. If there is no default Build ID, the result for the
+    // unversioned queue will be returned.
+    // (-- api-linter: core::0140::prepositions --)
+    temporal.api.taskqueue.v1.TaskQueueVersionGroup versions = 3;
```

Comment:
I'm wondering if the semantics we need for partition versions is the same as what we need for the whole task queue for user's need. e.g.:
- the default behavior as mentioned in the comment is useful for users but it's something that the root partition should calculate once and then pass the exact calculate version to the partitions when calling DescriveTaskQueuePartition
- the explicit unversioned oneof field may not be needed as internally we can support empty string in the build_ids field.
- maybe later we want to add new selections that make sense for internal use but not for end user. So tying these to may become limiting

Reply:
true I think the "default" should be calculated by root, the BuildIdList would just be passed as a list of strings as-is, and the unversioned could be converted into a string.

The only complicated one is "all_active." For "all_active," would it make sense for the root to first ask the partition "which of your physical queues are active," get a response, and then pass that list of build ids back to the partition? Vs passing "all_active" to the partition letting it calculate the list of relevant physical queues and use them in the same function, thus avoiding a round trip between the root and the partition.

Reply:
I'm thinking for now it can just be a single list. the default behavior would be if the list is empty it'll send all active. later if we want to support more modes internally, we can convert it to a oneof with more modes.

Reply:
speaking of, I'm clear on how to tell if there is a recent poller, but do you know how I can tell if a task queue has had a task added to it recently?

Reply:
on further thought, I think it would be nice to be able to pass just a list of build ids to `DescribeTaskQueuePartition` instead of the `TaskQueueVersionGroup`, I just don't see a way to avoid the extra round trip in the "all_active" scenario. Curious about your thoughts :)

Reply:
for recently added tasks, we can't definitely tell it right now. but that's fine for the time being, the definition of "active" can just mean "loaded" until we start tracking some sort of tracking like last task added timestamp.

about avoiding "all_active", we can say empty list means all active (more accurately, all loaded).

Reply:
I agree with all this. Have changed it to a `repeated string`